# Date: 13/07/2017
# Author: Tatyana Mozgacheva tmozgacheva@bcgsc.ca
# Description: This script generates a table of formulae's license in ORCA.
# In order to get the information about licenses, the git page is required.
# This page is located in either 'homepage' field or 'head' field in formulae's code.
# Procedure:
# 1) Run the Docker container using `docker run -ti ID_IMAGE` 
#    in the test server docker2 OR
#    Run a script /usr/local/bin/orca in the orca1 server; 
# 2) Inside the Docker containers, type:
#    *URL from the 'head' field: 
#    brew irb <<< "%w[$(brew list)]"'.map { |x| Formula[x].head}.compact.map {|v| v.url}'
#    and copy the output into 'urls' variable
#    *Name of formulae: 
#    brew irb <<< "%w[$(brew list)]"'.map { |x| Formula[x].head}.compact.map {|v| v.name}'
#    and copy the output into 'formulae' variable
#    *Name of formulae: 
#    brew irb <<< "%w[$(brew list)]"'.map { |x| Formula[x].name}'
#    and copy the output into 'formulae2' variable
#    *URL from the 'homepage' field: 
#    brew irb <<< "%w[$(brew list)]"'.map { |x| Formula[x].homepage}'
#    and copy the output into 'fhomepages' variable
# 3) Get rid of 'https://github.com', '.git' in order to get name of the repo.
# 4) Replace ??? by your github account and password in paste() functions
# 5) Run the following script:

install.packages("devtools")
install.packages("tidyjson")
install.packages("tidyverse")
install.packages("knitr")
install.packages("DT")

devtools::install_github("hrbrmstr/curlconverter")
library(devtools)
library(curlconverter)
library(jsonlite)
library(httr)

library(tidyverse)
library(knitr)
library(DT)

library(tidyjson)   

#Input from the container
urls <- c("/bcgsc/abyss", "/Sheikhizadeh/ACE", "/bigdatagenomics/adam", "/EvolBioInf/afra", "git-wip-us.apache.org/repos/asf/ant", "/apache/spark", "/bcgsc/arcs", "/juliema/aTRAM", "/DecodeGenetics/BamHash", "/macroevolution/bamm", "/genome/bam-readcount", "/pezmaster31/bamtools", "/statgen/bamUtil", "/tseemann/barrnap", "/GATB/bcalm", "/beagle-dev/beagle-lib", "/beast-dev/beast-mcmc", "/CompEvol/beast2", "/bedops/bedops", "/arq5x/bedtools2", "/BEETL/BEETL", "/lh3/bfc", "/lh3/bioawk", "/bcgsc/biobloom", "/evoldoers/biomake", "/PacificBiosciences/blasr", "/boostorg/boost", "/BenLangmead/bowtie", "/BenLangmead/bowtie2", "/barricklab/breseq", "/lh3/bwa", "anongit.freedesktop.org/git/cairo", "/marbl/canu", "/weizhongli/cdhit", "/infphilo/centrifuge", "/sanger-pathogens/circlator", "/xavierdidelot/ClonalFrameML", "/ivazquez/cloneHD", "cmake.org/cmake", "/miyagawa/cpanminus", "/thegenemyers/DALIGNER", "/thegenemyers/DAZZ_DB", "/thegenemyers/DEXTRACTOR", "/nh13/DWGSIM", "ftp://ftp.ncbi.nlm.nih.gov/entrez/entrezdirect/versions/current/edirect.tar.gz", "bitbucket.org/eigen/eigen", "/emacs-mirror/emacs", "/libexpat/libexpat", "/adarob/eXpress", "/agordon/fastx_toolkit", "/lh3/fermi", "/lh3/fermikit", "/lh3/fermi-lite", "anongit.freedesktop.org/git/fontconfig", "/ekg/freebayes", "/BoevaLab/FREEC", "/molpopgen/fwdpp", "/broadgsa/gatk-protected", "svn://gcc.gnu.org/svn/gcc/trunk", "/libgd/libgd", "/genometools/genometools", "/marbl/gingr", ":pserver:anonymous:@gnuplot.cvs.sourceforge.net:/cvsroot/gnuplot", "/gperftools/gperftools", "/arq5x/grabix", "/silnrsi/graphite", "/behdad/harfbuzz", "svn.janelia.org/eddylab/eddys/src/hmmer/trunk", "/lh3/htsbox", "/samtools/htslib", "ssl.icu-project.org/repos/icu/trunk/icu4c/", "/broadinstitute/IGV", "/igvteam/igv", "repo.or.cz/r/isl", "/sanger-pathogens/iva", "/gmarcais/Jellyfish", "/bioinformatics-centre/kaiju", "/TGAC/KAT", "git://genome-source.cse.ucsc.edu/kent", "/marekkokot/KMC", "/pmelsted/KmerStream", "/bcgsc/kollector", "last.cbrc.jp/last", "/atgreen/libffi", "git.assembla.com/phylogenetic-likelihood-library", "/glennrp/libpng", "/molpopgen/libsequence", "git://git.sv.gnu.org/libunwind", "git.gnome.org/browse/libxml2", "/mourisl/Lighter", "/mgymrek/lobstr-code", "/marbl/Mash", "/matplotlib/matplotlib", "/voutcn/megahit", "/marbl/MHAP", "/johnkerl/miller", "/ctSkennerton/minced", "/lh3/miniasm", "/lh3/minimap", "/hangelwen/miR-PREFeR", "/mothur/mothur", "mrbayes.svn.sourceforge.net/svnroot/mrbayes/trunk/", "/jts/nanopolish", "/lindenb/newicktools", "/tjunier/newick_utils", "/nextflow-io/nextflow", "/lmrodriguezr/nonpareil", "/bcgsc/ntCard", "/numpy/numpy", "/sequencing/NxTrim", "/xianyi/OpenBLAS", "/open-mpi/ompi", "/davidemms/OrthoFinder", "/neufeld/pandaseq", "git.gnome.org/browse/pango", "git.savannah.gnu.org/git/parallel", "/marbl/parsnp", "/NixOS/patchelf", "svn://vcs.exim.org/pcre/code/trunk", "perl5.perl.org/perl", "/chinchliff/phlawd", "/broadinstitute/pilon", "/hyattpd/Prodigal", "/tseemann/prokka", "/google/protobuf", "/lh3/psmc", "/python/cpython", "/python/cpython", "/khowe/quicktree", "svn.r-project.org/R/trunk", "/isovic/racon", "/TGAC/RAMPART", "/stamatak/standard-RAxML", "/sebhtml/ray", "/mourisl/Rcorrector", "/alexdobin/STAR", "/lh3/ropebwt2", "svn.ruby-lang.org/repos/ruby/trunk/", "/scipy/scipy", "/lh3/seqtk", "/SWI-Prolog/swipl-devel", "/trinityrnaseq/trinityrnaseq", "/karelzak/util-linux", "/ekg/vcflib", "/dzerbino/velvet", "/vim/vim", "chromium.googlesource.com/webm/libwebp", "git.savannah.gnu.org/git/wget")

formulae <- c("abyss", "ace-corrector", "adam", "afra", "ant", "apache-spark", "arcs", "atram", "bamhash", "bamm", "bam-readcount", "bamtools", "bamutil", "barrnap", "bcalm", "beagle", "beast", "beast2", "bedops", "bedtools", "beetl", "bfc", "bioawk", "biobloomtools", "biomake", "blasr", "boost", "bowtie", "bowtie2", "breseq", "bwa", "cairo", "canu", "cd-hit", "centrifuge", "circlator", "clonalframeml", "clonehd", "cmake", "cpanminus", "daligner", "dazz_db", "dextractor", "dwgsim", "edirect", "eigen", "emacs", "expat", "express", "fastx_toolkit", "fermi", "fermikit", "fermi-lite", "fontconfig", "freebayes", "freec", "fwdpp", "gatk", "gcc", "gd", "genometools", "gingr", "gnuplot", "gperftools", "grabix", "graphite2", "harfbuzz", "hmmer", "htsbox", "htslib", "icu4c", "igv", "igvtools", "isl", "iva", "jellyfish", "kaiju", "kat", "kent-tools", "kmc", "kmerstream", "kollector", "last", "libffi", "libpll", "libpng", "libsequence", "libunwind", "libxml2", "lighter", "lobstr", "mash", "matplotlib", "megahit", "mhap", "miller", "minced", "miniasm", "minimap", "mir-prefer", "mothur", "mrbayes", "nanopolish", "newicktools", "newick-utils", "nextflow", "nonpareil", "ntcard", "numpy", "nxtrim", "openblas", "open-mpi", "orthofinder", "pandaseq", "pango", "parallel", "parsnp", "patchelf", "pcre", "perl", "phlawd", "pilon", "prodigal", "prokka", "protobuf", "psmc", "python", "python3", "quicktree", "r", "racon", "rampart", "raxml", "ray", "rcorrector", "rna-star", "ropebwt2", "ruby", "scipy", "seqtk", "swi-prolog", "trinity", "util-linux", "vcflib", "velvet", "vim", "webp", "wget")

fhomepages <- c("sourceforge.net/projects/ngopt/", "abacas.sourceforge.io/", "www.bcgsc.ca/platform/bioinfo/software/abyss", "www.bcgsc.ca/platform/bioinfo/software/abyss-explorer", "/Sheikhizadeh/ACE", "/bigdatagenomics/adam", "/MikkelSchubert/adapterremoval", "/EvolBioInf/afra", "sourceforge.net/apps/mediawiki/amos/index.php?title=AMOS", "/EvolBioInf/andi", "ant.apache.org/", "spark.apache.org/", "mbio-serv2.mbioekol.lu.se/ARAGORN/", "/bcgsc/arcs", "argtable.sourceforge.io", "www.niehs.nih.gov/research/resources/software/biostatistics/art/index.cfm", "www.sanger.ac.uk/science/tools/artemis", "/smirarab/ASTRAL", "/juliema/aTRAM", "bioinf.uni-greifswald.de/augustus/", "www.gnu.org/software/autoconf", "www.gnu.org/software/automake/", "/DecodeGenetics/BamHash", "bamm-project.org", "/genome/bam-readcount", "/pezmaster31/bamtools", "genome.sph.umich.edu/wiki/BamUtil", "/tseemann/barrnap", "bbmap.sourceforge.io/", "/GATB/bcalm", "www.htslib.org/", "/beagle-dev/beagle-lib", "beast.bio.ed.ac.uk/", "www.beast2.org/", "/bedops/bedops", "/arq5x/bedtools2", "/BEETL/BEETL", "www.oracle.com/technology/products/berkeley-db/index.html", "www.oracle.com/technology/products/berkeley-db/index.html", "/lh3/bfc", "www.x.org/", "/lh3/bioawk", "www.bcgsc.ca/platform/bioinfo/software/biobloomtools/", "www.codamono.com/biointerchange/", "/evoldoers/biomake", "www.gnu.org/software/bison/", "/PacificBiosciences/blasr", "blast.ncbi.nlm.nih.gov/", "genome.ucsc.edu/FAQ/FAQblat.html", "sourceforge.net/projects/bless-ec/", "www.boost.org/", "bowtie-bio.sourceforge.io/", "bowtie-bio.sourceforge.io/", "barricklab.org/twiki/bin/view/Lab/ToolsBacterialGenomeResequencing", "busco.ezlab.org", "/lh3/bwa", "www.bzip.org/", "cairographics.org/", "canu.readthedocs.org/en/latest/", "seq.cs.iastate.edu/cap3.html", "capnproto.org/", "cd-hit.org", "korflab.ucdavis.edu/datasets/cegma/", "wgs-assembler.sourceforge.io/", "www.ccb.jhu.edu/software/centrifuge", "sourceforge.net/projects/ceruleanassembler/", "sanger-pathogens.github.io/circlator/", "clark.cs.ucr.edu/", "/xavierdidelot/ClonalFrameML", "www.sanger.ac.uk/science/tools/clonehd", "www.clustal.org/omega/", "www.clustal.org/clustal2/", "www.cmake.org/", "www.x.org/", "/miyagawa/cpanminus", "cole-trapnell-lab.github.io/cufflinks/", "curl.haxx.se/", "www.cytoscape.org/", "/thegenemyers/DALIGNER", "www.x.org/", "/thegenemyers/DAZZ_DB")
fhomepages <- c(fhomepages, c("/thegenemyers/DEXTRACTOR", "ab.inf.uni-tuebingen.de/software/diamond/", "www.bcgsc.ca/platform/bioinfo/software/dida", "www.broadinstitute.org/software/discovar/blog/", "www.x.org/", "www.x.org/", "www.x.org/", "minia.genouest.org/dsk/", "/nh13/DWGSIM", "www.ncbi.nlm.nih.gov/books/NBK179288/", "eigen.tuxfamily.org/", "www.gnu.org/software/emacs/", "emboss.sourceforge.io/", "www.csd.uwo.ca/~ilie/E-MEM/", "www.ebi.ac.uk/about/vertebrate-genomics/software/exonerate", "libexpat.github.io/", "bio.math.berkeley.edu/eXpress/", "faculty.virginia.edu/wrpearson/fasta/", "fastml.tau.ac.il/source.php", "www.bioinformatics.babraham.ac.uk/projects/fastqc/", "homes.cs.washington.edu/~dcjones/fastq-tools/", "microbesonline.org/fasttree/", "sourceforge.net/projects/fastuniq/", "hannonlab.cshl.edu/fastx_toolkit/", "/lh3/fermi", "/lh3/fermi2", "/lh3/fermikit", "/lh3/fermi-lite", "www.x.org/", "ccb.jhu.edu/software/FLASH/", "/westes/flex", "wiki.freedesktop.org/www/Software/fontconfig/", "www.x.org/", "sourceforge.net/projects/fqzcomp/", "/ekg/freebayes", "bioinfo.curie.fr/projects/freec/", "www.freetype.org/", "fsa.sourceforge.net/", "molpopgen.github.io/fwdpp/", "www.broadinstitute.org/software/gaemr/", "gatb.inria.fr/", "software.broadinstitute.org/gatk/", "gcc.gnu.org/", "libgd.github.io/", "www.gnu.org/software/gdbm/", "genome.crg.es/software/geneid/", "www.ebi.ac.uk/~birney/wise2/", "genometools.org/", "www.gnu.org/software/gettext/", "/marbl/harvest/blob/master/docs/content/gingr.rst", "developer.gnome.org/glib/", "ccb.jhu.edu/software/glimmerhmm/", "www.x.org/", "research-pub.gene.com/gmap/", "gmplib.org/", "www.gnuplot.info", "live.gnome.org/GObjectIntrospection", "/sparsehash/sparsehash", "/gperftools/gperftools", "/arq5x/grabix", "graphite.sil.org", "bitbucket.org/nsegata/graphlan/wiki/Home", "www.gnu.org/software/gsl/", "wiki.freedesktop.org/www/Software/HarfBuzz/", "www.hdfgroup.org/HDF5", "ccb.jhu.edu/software/hisat/", "ccb.jhu.edu/software/hisat2/", "www.bcgsc.ca/platform/bioinfo/software/hlaminer", "hmmer.janelia.org", "hmmer.janelia.org/", "/lh3/htsbox", "www.htslib.org/"))
fhomepages <- c(fhomepages, c("huttenhower.sph.harvard.edu/humann", "site.icu-project.org/", "i.cs.hku.hk/~alse/hkubrg/projects/idba/", "www.broadinstitute.org/software/igv", "www.broadinstitute.org/software/igv", "mathgen.stats.ox.ac.uk/impute/impute_v2.html", "eddylab.org/infernal/", "www.x.org/", "www.iqtree.org/", "isl.gforge.inria.fr", "/sanger-pathogens/iva", "www.oracle.com/technetwork/java/javase/downloads/index.html", "www.genome.umd.edu/jellyfish.html", "www.cbcb.umd.edu/software/jellyfish/", "www.ijg.org", "kaiju.binf.ku.dk/", "pachterlab.github.io/kallisto/", "/TGAC/KAT", "www.x.org/", "genome.ucsc.edu/", "kmacs.gobics.de/", "sun.aei.polsl.pl/kmc/", "kmergenie.bx.psu.edu/", "/pmelsted/KmerStream", "/bcgsc/kollector", "ccb.jhu.edu/software/kraken/", "last.cbrc.jp/", "/dpryan79/libBigWig", "www.x.org/", "thrysoee.dk/editline/", "libevent.org", "sourceware.org/libffi/", "www.x.org/", "www.x.org/", "www.x.org/", "multiprecision.org", "www.x.org/", "www.libpll.org/", "www.libpng.org/pub/png/libpng.html", "www.x.org/", "molpopgen.github.io/libsequence/", "www.x.org/", "libtiff.maptools.org/", "www.gnu.org/software/libtool/", "www.nongnu.org/libunwind/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "xmlsoft.org", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "pyyaml.org/wiki/LibYAML", "/mourisl/Lighter", "www.bcgsc.ca/platform/bioinfo/software/links", "lobstr.teamerlich.org", "/tothuhien/lsd-0.3beta", "www.lua.org/", "/arq5x/lumpy-sv", "www.gnu.org/software/m4", "bioweb.supagro.inra.fr/macse/", "mafft.cbrc.jp/alignment/software/index.html", "x.org/", "www.yandell-lab.org/software/maker.html", "colibread.inria.fr/software/mapsembler2/", "maq.sourceforge.io/", "/marbl/Mash", "matplotlib.org", "maven.apache.org/", "micans.org/mcl", "/voutcn/megahit", "meme-suite.org", "mercurial-scm.org/", "huttenhower.sph.harvard.edu/metaphlan", "/marbl/MHAP", "/johnkerl/miller", "/ctSkennerton/minced", "minia.genouest.org/", "/lh3/miniasm", "/lh3/minimap", "/hangelwen/miR-PREFeR", "dogma.ccbb.utexas.edu/mitofy/", "/tseemann/mlst", "www.mothur.org/", "www.mpfr.org/", "mrbayes.sourceforge.io/", "sourceforge.net/projects/mwt/", "mummer.sourceforge.io/", "www.drive5.com/muscle/"))
fhomepages <- c(fhomepages, c("dev.mysql.com/doc/refman/5.7/en/", "/jts/nanopolish", "ncl.sourceforge.io/", "www.gnu.org/software/ncurses/", "/lindenb/newicktools", "cegg.unige.ch/newick_utils", "www.nextflow.io/", "enve-omics.ce.gatech.edu/nonpareil", "www.novocraft.com/", "/bcgsc/ntCard", "www.numpy.org", "/sequencing/NxTrim", "www.ebi.ac.uk/~zerbino/oases/", "omabrowser.org/standalone/", "www.openblas.net/", "www.open-mpi.org/", "openssl.org/", "openssl.org/", "/davidemms/OrthoFinder", "abacus.gene.ucl.ac.uk/software/paml.html", "/neufeld/pandaseq", "www.pango.org/", "savannah.gnu.org/projects/parallel/", "/marbl/parsnp", "nixos.org/patchelf.html", "www2.math.su.se/PATHd8/", "www.pathvisio.org/", "www.pcre.org/", "www.exelixis-lab.org/pear", "www.perl.org/", "www.maths.otago.ac.nz/~dbryant/software.html", "www.phlawd.net/", "evolution.genetics.washington.edu/phylip.html", "www.atgc-montpellier.fr/phyml/", "blackrim.org/programs/phyutility/", "broadinstitute.github.io/picard/", "drive5.com/piler/", "drive5.com/pilercr/", "/broadinstitute/pilon/wiki", "cairographics.org/", "freedesktop.org/wiki/Software/pkg-config/", "zzz.bwh.harvard.edu/plink/", "sourceforge.net/projects/poamsa/", "www.x.org/", "prodigal.ornl.gov/", "www.vicbioinformatics.com/software.prokka.shtml", "www.bioinf.uni-leipzig.de/Software/proteinortho/", "/google/protobuf/", "/lh3/psmc", "www.python.org", "www.python.org/", "cab.spbu.ru/software/quast/", "www-hsc.usc.edu/~valouev/QuEST/QuEST.html"))
fhomepages <- c(fhomepages, c("www.sanger.ac.uk/resources/software/quicktree/", "www.genome.umd.edu/quorum.html", "www.r-project.org/", "/isovic/racon", "/TGAC/RAMPART", "www.x.org/", "rapsearch2.sourceforge.io/", "sco.h-its.org/exelixis/web/software/raxml/index.html", "denovoassembler.sourceforge.io/", "/mourisl/Rcorrector", "tiswww.case.edu/php/chet/readline/rltop.html", "iubio.bio.indiana.edu/soft/molbio/readseq/java/", "sourceforge.net/p/readsim/wiki/Home/", "www.repeatmasker.org/RepeatModeler.html", "www.x.org/", "www.x.org/", "www.repeatmasker.org/", "www.repeatmasker.org/RepeatModeler.html", "bix.ucsd.edu/repeatscout/", "www.x.org/", "www.repeatmasker.org/RMBlast.html", "www.cbs.dtu.dk/services/RNAmmer/", "/alexdobin/STAR", "/lh3/ropebwt2", "www.ruby-lang.org/", "www.htslib.org/", "www.scipy.org", "www.x.org/", "/lh3/seqtk", "www.sanger.ac.uk/science/tools/smalt-0", "korflab.ucdavis.edu/software.html", "bioinf.spbau.ru/spades/", "sqlite.org/", "www.swi-prolog.org/", "support.hdfgroup.org/HDF5/release/obtain5.html#extlibs", "www.bcgsc.ca/platform/bioinfo/software/tasr", "www.threadingbuildingblocks.org/", "www.ncbi.nlm.nih.gov/genbank/tbl2asn2/", "www.tcsh.org/", "tandem.bu.edu/trf/trf.html", "/lh3/trimadap", "www.usadellab.org/cms/?page=trimmomatic", "trinityrnaseq.github.io", "www.info-zip.org/pub/infozip/UnZip.html", "/karelzak/util-linux", "/ekg/vcflib", "www.ebi.ac.uk/~zerbino/velvet/", "www.x.org/", "www.tbi.univie.ac.at/~ronny/RNA/", "vim.sourceforge.io/", "developers.google.com/speed/webp/", "www.gnu.org/software/wget/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "www.x.org/", "tukaani.org/xz/", "zlib.net/"))


formulae2 <- c("a5", "abacas", "abyss", "abyss-explorer", "ace-corrector", "adam", "adapterremoval", "afra", "amos", "andi", "ant", "apache-spark", "aragorn", "arcs", "argtable", "art", "artemis", "astral", "atram", "augustus", "autoconf", "automake", "bamhash", "bamm", "bam-readcount", "bamtools", "bamutil", "barrnap", "bbtools", "bcalm", "bcftools", "beagle", "beast", "beast2", "bedops", "bedtools", "beetl", "berkeley-db", "berkeley-db@4", "bfc", "bigreqsproto", "bioawk", "biobloomtools", "biointerchange", "biomake", "bison", "blasr", "blast", "blat", "bless", "boost", "bowtie", "bowtie2", "breseq", "busco", "bwa", "bzip2", "cairo", "canu", "cap3", "capnp", "cd-hit", "cegma", "celera-assembler", "centrifuge", "cerulean", "circlator", "clark", "clonalframeml", "clonehd", "clustal-omega", "clustal-w", "cmake", "compositeproto", "cpanminus", "cufflinks", "curl", "cytoscape", "daligner", "damageproto", "dazz_db", "dextractor", "diamond", "dida", "discovar", "dmxproto", "dri2proto", "dri3proto", "dsk")
formulae2 <- c(formulae2, c("dwgsim", "edirect", "eigen", "emacs", "emboss", "e-mem", "exonerate", "expat", "express", "fasta", "fastml", "fastqc", "fastq-tools", "fasttree", "fastuniq", "fastx_toolkit", "fermi", "fermi2", "fermikit", "fermi-lite", "fixesproto", "flash", "flex", "fontconfig", "fontsproto", "fqzcomp", "freebayes", "freec", "freetype", "fsa", "fwdpp", "gaemr", "gatb", "gatk", "gcc", "gd", "gdbm", "geneid", "genewise", "genometools", "gettext", "gingr", "glib", "glimmerhmm", "glproto", "gmap-gsnap", "gmp", "gnuplot", "gobject-introspection", "google-sparsehash", "gperftools", "grabix", "graphite2", "graphlan", "gsl", "harfbuzz", "hdf5", "hisat", "hisat2", "hlaminer", "hmmer", "hmmer2", "htsbox", "htslib", "humann2", "icu4c", "idba", "igv", "igvtools", "impute2", "infernal", "inputproto", "iqtree", "isl", "iva", "jdk", "jellyfish", "jellyfish-1.1", "jpeg", "kaiju", "kallisto", "kat", "kbproto", "kent-tools", "kmacs", "kmc", "kmergenie", "kmerstream", "kollector", "kraken", "last", "libbigwig", "libdmx", "libedit", "libevent", "libffi", "libfontenc"))
formulae2 <- c(formulae2, c("libfs", "libice", "libmpc", "libpciaccess", "libpll", "libpng", "libpthread-stubs", "libsequence", "libsm", "libtiff", "libtool", "libunwind", "libx11", "libxau", "libxaw", "libxcb", "libxcomposite", "libxcursor", "libxdamage", "libxdmcp", "libxext", "libxfixes", "libxfont", "libxft", "libxi", "libxinerama", "libxkbfile", "libxml2", "libxmu", "libxpm", "libxrandr", "libxrender", "libxres", "libxscrnsaver", "libxshmfence", "libxt", "libxtst", "libxv", "libxvmc", "libxxf86dga", "libxxf86vm", "libyaml", "lighter", "links-scaffolder", "lobstr", "lsd", "lua", "lumpy-sv", "m4", "macse", "mafft", "makedepend", "maker", "mapsembler2", "maq", "mash", "matplotlib", "maven", "mcl", "megahit", "meme", "mercurial", "metaphlan", "mhap", "miller", "minced", "minia", "miniasm", "minimap", "mir-prefer", "mitofy", "mlst", "mothur", "mpfr", "mrbayes", "multi-worm-tracker", "mummer", "muscle", "mysql", "nanopolish", "ncl", "ncurses", "newicktools", "newick-utils", "nextflow", "nonpareil", "novoalign", "ntcard", "numpy", "nxtrim", "oases", "oma", "openblas", "open-mpi", "openssl", "openssl@1.1", "orthofinder", "paml", "pandaseq", "pango", "parallel", "parsnp", "patchelf"))
formulae2 <- c(formulae2, c("pathd8", "pathvisio", "pcre", "pear", "perl", "phipack", "phlawd", "phylip", "phyml", "phyutility", "picard-tools", "piler", "pilercr", "pilon", "pixman", "pkg-config", "plink", "poa", "presentproto", "prodigal", "prokka", "proteinortho", "protobuf", "psmc", "python", "python3", "quast", "quest", "quicktree", "quorum", "r", "racon", "rampart", "randrproto", "rapsearch2", "raxml", "ray", "rcorrector", "readline", "readseq", "readsim", "recon", "recordproto", "renderproto", "repeatmasker", "repeatmodeler", "repeatscout", "resourceproto", "rmblast", "rnammer", "rna-star", "ropebwt2", "ruby", "samtools", "scipy", "scrnsaverproto", "seqtk", "smalt", "snap", "spades", "sqlite", "swi-prolog", "szip", "tasr", "tbb", "tbl2asn", "tcsh", "trf", "trimadap", "trimmomatic", "trinity", "unzip", "util-linux", "vcflib", "velvet", "videoproto", "viennarna", "vim", "webp", "wget", "xcmiscproto", "xextproto", "xf86bigfontproto", "xf86dgaproto", "xf86driproto", "xf86vidmodeproto", "xineramaproto", "xorg", "xorg-protocols", "xproto", "xtrans", "xz", "zlib"))

# Type of license from 'head' URLs
vectorr <- list()
df <- tibble

for (i in 1:length(urls)) {
  curlExample <- paste("curl -X GET -u ???:??? https://api.github.com/repos",urls[i],"/license", sep = "")
  resp <- make_req(straighten(curlExample))
  tJSON <- toJSON(content(resp[[1]](), as="parsed"), pretty=TRUE)
  dJSON <- jsonlite::fromJSON(tJSON, simplifyDataFrame = TRUE)
  vectorr <- c(vectorr, list(dJSON$license$key))
}

tibt <- as_data_frame(setNames(list(formulae,vectorr), c("Formula","Type")))

# Type of license from 'homepage' URLs
vectorr <- list()
df <- tibble

for (i in 1:length(fhomepages)) {
  curlExample <- paste("curl -X GET -u ???:??? https://api.github.com/repos",fhomepages[i],"/license", sep = "")
  resp <- make_req(straighten(curlExample))
  tJSON <- toJSON(content(resp[[1]](), as="parsed"), pretty=TRUE)
  dJSON <- jsonlite::fromJSON(tJSON, simplifyDataFrame = TRUE)
  vectorr <- c(vectorr, list(dJSON$license$key))
  
}

tibt2 <- as_data_frame(setNames(list(formulae2,vectorr), c("Formula","Type")))

# Combine
full_join <- full_join(tibt2, tibt, by = "Formula")

full_join$Type.x[full_join$Type.x=="NULL"] <- 'no git page'
full_join$Type.y[full_join$Type.y=="NULL"] <- 'no git page'
unite(full_join, new, full_join$Type.x:full_join$Type.y, sep='')

full_join$comb <- with(full_join, ifelse (full_join$Type.x == 'no git page',full_join$Type.y, full_join$Type.x))


pareto.chart(table(c(unlist((full_join$comb)))), main = "ORCA licenses",col=topo.colors(6))

